cmd/ocm-backplane/login/login.go

```diff
@@ -119,47 +130,29 @@ func runLogin(cmd *cobra.Command, argv []string) (err error) {
 	}
 	logger.Debugf("Backplane Config File Contains: %v \n", bpConfig)
 
+	// login to the cluster based on login type
 	logger.Debugf("Extracting Backplane Cluster ID")
-	// Currently go-pagerduty pkg does not include incident id validation.
-	if args.pd != "" {
-		if bpConfig.PagerDutyAPIKey == "" {
-			return fmt.Errorf("please make sure the PD API Key is configured correctly in the config file")
-		}
-		pdClient, err := pagerduty.NewWithToken(bpConfig.PagerDutyAPIKey)
+	switch loginType {
+	case LoginTypePagerduty:
+		info, err := getClusterInfoFromPagerduty(bpConfig)
+		clusterKey = info.ClusterID
+		elevateReason = info.WebURL
 		if err != nil {
-			return fmt.Errorf("could not initialize the client: %w", err)
-		}
-		if strings.Contains(args.pd, "/incidents/") {
-			incidentID := args.pd[strings.LastIndex(args.pd, "/")+1:]
-			clusterKey, err = pdClient.GetClusterIDFromIncident(incidentID)
-			if err != nil {
-				return err
-			}
-		} else {
-			clusterKey, err = pdClient.GetClusterIDFromIncident(args.pd)
-			if err != nil {
-				return err
-			}
+			return err
 		}
-	}
 
-	// Get the cluster ID only if it hasn't been populated by PagerDuty.
-	if len(argv) == 1 {
-		// if explicitly one cluster key given, use it to log in.
+	case LoginTypeClusterID:
 		logger.Debugf("Cluster Key is given in argument")
 		clusterKey = argv[0]
-		logger.WithField("Search Key", clusterKey).Debugln("Finding target cluster")
-
-	} else if len(argv) == 0 && args.pd == "" {
-		// if no args given, try to log into the cluster that the user is logged into
-		logger.Debugf("Finding Clustrer Key from current cluster")
-		clusterInfo, err := utils.DefaultClusterUtils.GetBackplaneClusterFromConfig()
+	case LoginTypeExistingKubeConfig:
+		clusterKey, err = getClusterIDFromExistingKubeConfig()
 		if err != nil {
 			return err
 		}
-		clusterKey = clusterInfo.ClusterID
-		logger.Debugf("Backplane Cluster Infromation data extracted: %+v\n", clusterInfo)
+	default:
+		return fmt.Errorf("please make sure the PD API Key is configured correctly in the config file")
```

Comment:
If we get here, means the loginType cannot be detected? Then it would be better to update the error message here, and put this error within the `case LoginTypePagerduty`?

Reply:
done